MiniApp/Classes/MiniAppDownloader.swift

```diff
@@ -22,15 +22,26 @@ class MiniAppDownloader {
     }
 
     func download(appId: String, versionId: String, completionHandler: @escaping (Result<URL, Error>) -> Void) {
-        let miniAppStoragePath = FileManager.getMiniAppDirectory(with: appId, and: versionId)
-        if miniAppStatus.isDownloaded(key: "\(appId)/\(versionId)") {
+        let miniAppStoragePath = FileManager.getMiniAppVersionDirectory(with: appId, and: versionId)
+        if miniAppStatus.isDownloaded(appId: appId, versionId: versionId) {
             completionHandler(.success(miniAppStoragePath))
             return
         }
         self.manifestDownloader.fetchManifest(apiClient: self.miniAppClient, appId: appId, versionId: versionId) { (result) in
             switch result {
             case .success(let responseData):
-                self.downloadMiniApp(urls: responseData.manifest, to: miniAppStoragePath, completionHandler: completionHandler)
+                self.downloadMiniApp(urls: responseData.manifest, to: miniAppStoragePath) { downloadResult in
+                    switch downloadResult {
+                    case .success:
+                        if let error = self.miniAppStorage.cleanVersions(for: appId, differentFrom: versionId, status: self.miniAppStatus) {
+                            print("WARNING: MiniAppDownloader could not delete previously downloaded versions for appId \(appId) (\(error))")
```

Comment:
Should you use the `Logger` class for this instead?

Reply:
we talked about removing it with @leorjoseph so I did not want to use it, but well, seems a good idea if we are keeping it